tests/orqviz/scans/scans_test.py

```diff
@@ -196,6 +196,44 @@ def test_perform_2D_scan_in_5D_space():
             )
 
 
+def test_perform_2D_scan_with_ND_array():
+    param_shape = (9, 4)
+    origin = np.zeros(param_shape)
+    direction_x = get_random_normal_vector(param_shape)
+    direction_y = get_random_orthonormal_vector(direction_x)
+    end_points_x = (-np.pi, np.pi)
+    end_points_y = (-np.pi / 2, np.pi / 2)
+    n_steps_x = 100
+    n_steps_y = 10
+
+    scan_2d = perform_2D_scan(
```

Comment:
Even if this provides a useful feature (and I guess it does, as otherwise you wouldn't implement that), it's pretty confusing, as I have no idea how I should interpret direction being an array.

Reply:
It is still a vector in `n x m` space. Until now, we would have to flatten the array, just to reshape it back later on. This is what I want to avoid.

Reply:
Wait, isn't a " vector in `n x m` space" a matrix?

Did you meant this for the case, where the space is actually N-dimensional, but there is some natural representation of the parameters so that `n x m = N`, e.g. for QAOA that would be `n=n_layers` and `m=2`?

So even though under the hood we still operate on the N-dimensional objects, at least now the interface is cleaner and we don't need to reshape it every time?

If that's the case I think calling 2-dimensional objects "vectors" is still quite confusing, even if mathematically correct.

Reply:
Any comments here @MSRudolph ?

Reply:
How else would you call it? `ParameterTensor`? I feel like it is totally fine to emphasize the norm being `(n, )` shapes, but allowing different ones.